src/distilabel/steps/task/base.py

```diff
@@ -85,3 +85,7 @@ def process(self, inputs: StepInput) -> "StepOutput":  # type: ignore
             input["model_name"] = self.llm.model_name
             outputs.append(input)
         yield outputs
+
+
+class GeneratorTask(GeneratorStep, Task):
```

Comment:
Hmm but if the `Task` implements `process` and `load` which are shared for both `Task` and `GeneratorTask` what's the purpose on creating `_Task`? Because what you mentioned above implies `_Task` -> `Task(Step, _Task)` -> `GeneratorTask(GeneratorStep, Task)`, is that correct? Othewise what you mentioned implies that `_Task` contains everything including `load` and `process` and both `Task` and `GeneratorTask` inherit from `_Task` and either `Step` or `GeneratorStep`, and then `pass`.